Migrate hardcoded values into Nala tokens

aguscruiz opened the pull request:
resolves https://github.com/brave/brave-browser/issues/41655

- Migrate menu background colors
- Migrate container and border colors
- Migrate general background color

<!-- Add brave-browser issue below that this PR will resolve -->
Resolves

<!-- CI-related labels that can be applied to this PR:
* CI/run-audit-deps (1) - check for known npm/cargo vulnerabilities (audit_deps)
* CI/run-network-audit (1) - run network-audit
* CI/run-upstream-tests - run Chromium unit and browser tests on Linux and Windows (otherwise only on Linux)
* CI/run-linux-arm64, CI/run-macos-x64, CI/run-windows-arm64, CI/run-windows-x86 - run builds that would otherwise be skipped
* CI/skip - do not run CI builds (except noplatform)
* CI/skip-linux-x64, CI/skip-android, CI/skip-macos-arm64, CI/skip-ios, CI/skip-windows-x64 - skip CI builds for specific platforms
* CI/skip-upstream-tests - do not run Chromium unit, or browser tests (otherwise only on Linux)
* CI/skip-all-linters - do not run presubmit and lint checks
* CI/storybook-url (1) - deploy storybook and provide a unique URL for each build

(1) applied automatically when some files are changed (see: https://github.com/brave/brave-core/blob/master/.github/labeler.yml)
-->

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Page structure should now look more closer to the designs in Figma https://www.figma.com/design/IXNCYwXmMWaF6u7Pdv1pOW/Desktop-settings---Nala?node-id=1320-42369&t=9JeU7mJwEKd7K9CT-1
- Light and dark mode colors should resemble Figma designs too https://www.figma.com/design/IXNCYwXmMWaF6u7Pdv1pOW/Desktop-settings---Nala?node-id=1621-12827&t=9JeU7mJwEKd7K9CT-1
- Menu should be sticky on the left side of the page, always against the edge of the screen.
- The content should be centered in the middle of the remaining space
- Sub pages should look close to the designs instead of covering the whole content with a color background https://www.figma.com/design/IXNCYwXmMWaF6u7Pdv1pOW/Desktop-settings---Nala?node-id=505-37563&t=9JeU7mJwEKd7K9CT-1

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://build-artifacts.brave.com/brave-browser-build-pr/migrate-settings-to-nala/storybook.html)

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_menu.ts:
```diff
@@ -54,17 +54,16 @@ RegisterStyleOverride(
+        margin: 0 !important;
         max-height: calc(100vh - 56px - (var(--brave-settings-menu-margin-v) * 2) - (var(--brave-settings-menu-padding) * 2));
         min-width: 172px;
+        max-width:250px;
```
```suggestion
        max-width: 250px;
```

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_menu.ts:
```diff
@@ -54,17 +54,16 @@ RegisterStyleOverride(
-        background-color: #fff;
         overflow-y: auto;
         padding: 24px !important;
-        box-shadow: var(--leo-effect-elevation-01);
```
we don't need the shadow anymore?

aguscruiz replied:
No because it's now docked to the side, no need for shadow

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_menu.ts:
```diff
@@ -152,7 +150,8 @@ RegisterStyleOverride(
 
         a:hover, cr-icon:hover {
-          color: #A6ABE9 !important;
+          color: var(--leo-color-icon-interactive) !important;
+          --iron-icon-fill-color: var(--leo-color-icon-interactive) !important;
```
nit: variable declarations should come first

```suggestion
          --iron-icon-fill-color: var(--leo-color-icon-interactive) !important;
          color: var(--leo-color-icon-interactive) !important;
```

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_menu.ts:
```diff
@@ -190,12 +189,12 @@ RegisterStyleOverride(
       .brave-about-graphic {
         flex: 0;
-        flex-basis: 30%;
+        flex-basis: var(--leo-spacing-3xl);
```
what does this do?

aguscruiz replied:
Before, it was setting the padding between the logo and the text in a % of the whole menu width, super weird behavior. I set it so it's fixed and keeps the same alignment as the nav buttons

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_section.ts:
```diff
@@ -11,11 +11,25 @@ RegisterStyleOverride(
     <style include="settings-shared">
       #header .title {
-        font-weight: 500 !important;
+        font-weight: 600 !important;
+        font-size: var(--leo-typography-heading-h3-font-size) !important;
```
I take it we can't just use:

```suggestion
        font: var(--leo-heading-h3) !important;
```

for some reason?

aguscruiz replied:
Ohhhh yeah we could, forgot about that

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_subpage.ts:
```diff
@@ -0,0 +1,44 @@
+    <style include="settings-shared">
+    :host{
+        min-height: auto !important;
+        display:inline;
+        position:relative;
```
nit:
```suggestion
        display: inline;
        position: relative;
```

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_subpage.ts:
```diff
@@ -0,0 +1,44 @@
+  'settings-subpage',
+  html`
+    <style include="settings-shared">
+    :host{
```
nit:

```suggestion
    :host {
```

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_subpage.ts:
```diff
@@ -0,0 +1,44 @@
+        background-color: var(--leo-color-page-background) !important;
+        box-shadow: none !important;
+    }
+    slot{
```
nit:
```suggestion
    slot {
```

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_subpage.ts:
```diff
@@ -0,0 +1,44 @@
+        font-size: var(--leo-typography-heading-h3-font-size) !important;
+    }
+    
+      </style>
```
nit:
```suggestion
  </style>
```

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_subpage.ts:
```diff
@@ -0,0 +1,44 @@
+    
+      </style>
+  `
+)
```
nit:
```suggestion
)

```

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_ui.ts:
```diff
@@ -49,27 +50,36 @@ RegisterStyleOverride(
       cr-drawer settings-menu::part(header) {
         display: none;
       }
+      .cr-container-shadow{
+        display:none !important;
```
nit:
```suggestion
      .cr-container-shadow {
        display: none !important;
```

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_ui.ts:
```diff
@@ -49,27 +50,36 @@ RegisterStyleOverride(
       #left {
-        /* fixed size menu */
-        flex: 0 0 calc(var(--settings-menu-width) + var(--brave-settings-menu-margin) * 2)  !important;
+        max-width:250px;
```
nit:
```suggestion
        max-width: 250px;
```

fallaciousreasoning commented on browser/resources/settings/brave_overrides/settings_ui.ts:
```diff
@@ -49,27 +50,36 @@ RegisterStyleOverride(
+      }
+      @media (max-width: 980px) {
+        #main {
+          margin:0;
+          border-radius:0;
+          min-height:100%;
```
nit:
```suggestion
          margin: 0;
          border-radius: 0;
          min-height: 100%;
```

fallaciousreasoning commented on browser/resources/settings/brave_wallet_page/wallet_networks_list.html:
```diff
@@ -1,15 +1,21 @@
   .default-network {
     font-weight: bold;
   }
+  .settings-box{
+    padding: 0 var(--cr-section-padding);
+  }
+  .settings-row{
+    padding: var(--leo-spacing-m) var(--leo-spacing-2xl);
+  }
```
```suggestion
  .settings-box {
    padding: 0 var(--cr-section-padding);
  }
  .settings-row {
    padding: var(--leo-spacing-m) var(--leo-spacing-2xl);
  }
```

fallaciousreasoning commented on browser/resources/settings/brave_wallet_page/wallet_networks_subpage.html:
```diff
@@ -1,4 +1,5 @@
 <style include="cr-shared-style" >
+ 
```
nit: revert

```suggestion
```

fallaciousreasoning commented on ui/webui/resources/br_elements/br_toolbar/br_toolbar.html:
```diff
@@ -38,21 +38,20 @@
-    font-weight: 300;
-    letter-spacing: 0.6px;
-    /* hide until fonts are loaded */
+    font-weight: 500;
```
We don't seem to have a `leo` variable with this weight - should this maybe be one of
1. --leo-font-large-link: 400 16px/24px Inter Variable;
2.  --leo-font-large-regular: 400 16px/24px Inter Variable;

```suggestion
    font: var(--leo-font-the-right-one);
```

aguscruiz replied:
Awesome, makes sense

fallaciousreasoning commented on ui/webui/resources/br_elements/br_toolbar/br_toolbar.html:
```diff
@@ -38,21 +38,20 @@
     text-decoration: none;
-    transition: opacity 100ms ease-out;
+    transition: background 100ms ease-out;
+    height:44px;
```
nit:
```suggestion
    height: 44px;
```

fallaciousreasoning reviewed (commented):
Wow, this looks amazing @aguscruiz. Great work!

My only comments really are around code style - this is super impressive, and the settings page looks much better after this.

aguscruiz commented:
@fallaciousreasoning this is ready for being merged I think. I addressed your suggestions. There are more styles in subpages, like hover states on rows and stuff, but it's requiring me to create more template pages. So I'm just gonna leave it like this for now.

goodov reviewed (approved):
`source.gni` lgtm

fallaciousreasoning commented:
@aguscruiz presubmit is failing - can you run `npm run presubmit -- --fix`

brave-builds commented:
Released in [v1.73.32](https://github.com/brave/brave-browser/releases/tag/v1.73.32)
